components/dashboard/src/start/StartWorkspace.tsx

```diff
@@ -21,10 +22,51 @@ const sessionId = v4();
 const WorkspaceLogs = React.lazy(() => import('../components/WorkspaceLogs'));
 
 export interface StartWorkspaceProps {
-  workspaceId: string;
+  workspaceId: string,
+  runsInIFrame: boolean,
+  /**
+   * This flag is used to break the autostart-cycle explained in https://github.com/gitpod-io/gitpod/issues/8043
+   */
+  dontAutostart: boolean,
+}
+
+export function parseProps(workspaceId: string, search?: string): StartWorkspaceProps {
+  const params = parseParameters(search);
+  const notFound = !!(params && params["not_found"]);
+  const runsInIFrame = window.top !== window.self;
+  return {
+    workspaceId,
+    runsInIFrame: window.top !== window.self,
+    // Either:
+    //  - not_found: we were sent back from a workspace cluster/IDE URL where we expected a workspace to be running but it wasn't because either:
+    //    - this is a (very) old tab and the workspace already timed out
+    //    - due to a start error our workspace terminated very quickly between:
+    //      a) us being redirected to that IDEUrl (based on the first ws-manager update) and
+    //      b) our requests being validated by ws-proxy
+    //  - runsInIFrame (IDE case):
+    //    - we assume the workspace has already been started for us
+    //    - we don't know it's instanceId
+    dontAutostart: notFound || runsInIFrame,
+  }
+}
+
+function parseParameters(search?: string) {
+  try {
+    if (search === undefined) {
+      return undefined;
+    }
+    return queryString.parse(search, {parseBooleans: true});
```

Comment:
this looks like a vulnerability in the making. Considering we're making the result of this parse operation props on an element, I'd be weary one could use this for some XSS/prototype poisoning/CSS injection type of attack.

We should make sure we don't allow props we don't expect.

Reply:
Awh, yes! That got removed with one of the simplification steps above - will re-add. :pray:

Reply:
Looking at the code again it we did not assign the result right away anymore (that was a previous version).
To avoid confusion - and minimize risk over future mis-interpretation this now looks like this:
```
function parseParameters(search?: string): { notFound?: boolean } {
  try {
    if (search === undefined) {
      return {};
    }
    const params = queryString.parse(search, {parseBooleans: true});
    const notFound = !!(params && params["not_found"]);
    return {
      notFound,
    };
  } catch (err) {
    console.error("/start: error parsing search params", err);
    return {};
  }
}
```

Reply:
🚀  Thanks :)